jwst/source_catalog/source_catalog_step.py

```diff
@@ -46,17 +47,14 @@ def process(self, input):
 
             self.log.info('Detected {0} sources'.format(len(catalog)))
 
-            old_suffixes = ['i2d']
-            output_dir = self.search_attr('output_dir')
-            cat_filepath = replace_suffix_ext(model.meta.filename,
-                                              old_suffixes, 'cat',
-                                              output_ext='ecsv',
-                                              output_dir=output_dir)
-            catalog.write(cat_filepath, format='ascii.ecsv', overwrite=True)
-            self.log.info('Wrote source catalog: {0}'
-                          .format(cat_filepath))
-            model.meta.source_catalog.filename = cat_filepath
+            if self.save_results:
+                cat_filepath = self.make_output_path()
```

Comment:
If you update this line to:
`cat_filepath = self.make_output_path(basepath=model.meta.filename)`

I think it works, @stscieisenhamer should verify that is valid for full filepath names toos

```
2018-06-08 13:46:47,208 - stpipe.SourceCatalogStep - INFO - Detected 49 sources
2018-06-08 13:46:47,257 - stpipe.SourceCatalogStep - INFO - Wrote source catalog: niriss_F200W_image_position1_cat.ecsv
```